client/src/app/gui/rtp_streams_plot.rs

```diff
@@ -37,154 +42,253 @@ impl Display for SettingsXAxis {
-                            }
-                            RawTimestamp => { (packet.timestamp.as_secs_f64(), stream_ix as f64) }
-                            SequenceNumer => { (rtp_packet.sequence_number as f64, stream_ix as f64) }
-                        };
-
-                        on_hover.push_str(&format!("x = {} [{}]\n", x, self.settings_x_axis));
-                        let point = Points::new([x, y]).name(on_hover).color(color).radius(1.5);
-
-                        points.push(point);
-                        points_xy.push((x, y));
-                    });
-                });
-
-                let plot = Plot::new("halo")
-                    .show_background(false)
-                    .show_axes([true, false])
-                    .label_formatter(|name, _value| name.to_string())
-                    .view_aspect(2.0);
-
-                if self.requires_reset {
-                    plot.reset().show(ui, |plot_ui| {
-                        for point in points {
-                            plot_ui.points(point);
-                        }
-                    });
-                } else {
-                    plot.show(ui, |plot_ui| {
-                        for point in points {
-                            plot_ui.points(point);
-                        }
-                    });
+            ui.collapsing("Settings", |ui| {
+                self.options_ui(ui);
+            });
+            self.plot_ui(ui);
+        });
+    }
+
+    fn options_ui(&mut self, ui: &mut Ui) {
+        ui.horizontal(|ui| {
+            ui.label("X axis value:");
+            SettingsXAxis::all().into_iter().for_each(|setting| {
+                if ui
+                    .radio(setting == self.settings_x_axis, setting.to_string())
+                    .clicked()
+                {
+                    self.settings_x_axis = setting;
+                    self.requires_reset = true;
                 }
-                self.requires_reset = false
-            })
+            });
         });
+        ui.horizontal_wrapped(|ui| {
+            let streams = &self.streams.borrow().streams;
+            let ssrcs: Vec<_> = streams.keys().collect();
+
+            ui.label("Toggle streams: ");
+            ssrcs.iter().for_each(|&ssrc| {
+                let selected = is_stream_visible(&mut self.streams_visibility, *ssrc);
+                let resp = ui.checkbox(
+                    selected,
+                    streams.get(ssrc).unwrap().display_name.to_string(),
+                );
+                if resp.clicked() {
+                    self.requires_reset = true
+                }
+            });
+        });
+        ui.vertical(|ui| {
+            ui.add_space(10.0);
+        });
+    }
+
+    fn plot_ui(&mut self, ui: &mut Ui) {
+        let number_of_packets = self.streams.borrow().packets.len();
+        if self.last_packets_len != number_of_packets || self.requires_reset {
+            self.refresh_points();
+        }
```

Comment:
I'm wondering if we shouldn't base this condition on number of RTP packets specifically, not all packets? Imagine that I used "parse as RTP" option on some packet that wasn't recognized as RTP. Number of packets is the same, but number of RTP packets changed. That would require filtering the packets list on every frame, which also kinda sucks.

I think we can leave it as it is, but add a `FIXME:` comment with what I wrote above, more or less.

Reply:
```rust
    fn number_of_rtp_packets(&mut self) -> usize {
        self
            .streams
            .borrow()
            .streams
            .values()
            .map(|stream| stream.rtp_packets.len())
            .sum()
    }
```

Reply:
Why wait? LET'S DO IT!